Sources/MapboxNavigation/WayNameLabel.swift

```diff
@@ -0,0 +1,154 @@
+        let shieldRepresentation = representation?.shield
+        let legacyRoadShieldImage = spriteRepository.legacyRoadShieldImage(from: representation?.legacyCacheKey)
+        
+        // For US state road, use the legacy shield first, then fall back to use the generic shield icon.
+        // The shield name for US state road is `circle-white` in Streets source v8 style.
+        if let shieldRepresentation = shieldRepresentation,
+           shieldRepresentation.name == "circle-white",
+           let legacyRoadShieldImage = legacyRoadShieldImage {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: legacyRoadShieldImage)
+            
+            return
+        }
+        
+        // For non US state road, use the generic shield icon first, then fall back to use the legacy shield.
+        if let roadShieldImage = spriteRepository.roadShieldImage(from: shieldRepresentation) {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: roadShieldImage,
+                              roadShieldText: shieldRepresentation?.text,
+                              roadShieldTextColor: shieldRepresentation?.textColor)
+            
+            return
+        }
+        
+        // In case if legacy shield icon is available - use it.
+        if let legacyRoadShieldImage = legacyRoadShieldImage {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: legacyRoadShieldImage)
+            
+            return
+        }
+        
+        // In case if neither generic nor legacy shield images are available - show only road name.
+        text = roadName
+    }
+    
+    private func setAttributedText(roadName: String,
+                                   roadShieldImage: UIImage,
+                                   roadShieldText: String? = nil,
+                                   roadShieldTextColor: String? = nil) {
+        let roadShieldAttributedString = roadShield(from: roadShieldImage,
+                                                    text: roadShieldText,
+                                                    textColor: roadShieldTextColor)
+        
+        let roadShieldAndNameAttributedString = NSMutableAttributedString()
+        // Road shield image is always drawn in front of the road name.
+        roadShieldAndNameAttributedString.append(roadShieldAttributedString)
+        roadShieldAndNameAttributedString.append(NSAttributedString(string: " \(roadName)"))
+        attributedText = roadShieldAndNameAttributedString
+    }
+    
+    private func roadShield(from image: UIImage,
+                            text: String? = nil,
+                            textColor: String? = nil) -> NSAttributedString {
+        let attachment = ShieldAttachment()
+        // Shield attachment should use similar font that is currently used in `WayNameLabel`.
+        attachment.font = font
```

Comment:
Previously, while placing shield image incorrect font was used (default one). This caused improper alignment of generic shield image and road name label. Applying correct for attachment fixes this.

Reply:
Confirmed that we’re setting a non-default font in every occurrence of ShieldAttachment in InstructionPresenter.